test/Altinn.AccessManagement.Tests/Mocks/Altinn2RightsClientMock.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Altinn.AccessManagement.Core.Clients.Interfaces;
+using Altinn.AccessManagement.Core.Models;
+using Altinn.AccessManagement.Core.Models.ResourceRegistry;
+using Altinn.AccessManagement.Integration.Clients;
+using Altinn.AccessManagement.Models;
+using static Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext;
+
+namespace Altinn.AccessManagement.Tests.Mocks
+{
+    /// <summary>
+    /// Mock class for <see cref="IResourceRegistryClient"></see> interface
+    /// </summary>
+    public class Altinn2RightsClientMock : IAltinn2RightsClient
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Altinn2RightsClientMock"/> class
+        /// </summary>
+        public Altinn2RightsClientMock()
+        {
+        }
+
+        /// <inheritdoc/>
+        public async Task<DelegationCheckResponse> PostDelegationCheck(int userId, string partyId, string serviceCode, string serviceEditionCode)
```

Comment:
can remove the async to fix code smell since mock implementations don't have any await async operations